app/controllers/registrations_controller.rb

```diff
@@ -0,0 +1,8 @@
+class RegistrationsController < Devise::RegistrationsController
+
+  protected
+
+  def update_resource(user, params)
```

Comment:
what is `user` here?

Reply:
it was `resource` in the docs. I'm not sure how the code worked before. I'm updating the code within the method, so that the receiver matches the argument:

`  def update_resource(user, params)`
`    user.update_without_password(params)`
 ` end`

Reply:
great! I think Devise still uses a gem called `InheritedResources`, which adds a `resource` method for controllers and views to use. That's probably why what you had worked.